openedx/core/djangoapps/user_api/accounts/views.py

```diff
@@ -202,3 +205,23 @@ def partial_update(self, request, username):
             )
 
         return Response(account_settings)
+
+
+class AccountDeactivationView(APIView):
+    """
+    Account deactivation viewset. Currently only supports POST requests.
+    Only admins can deactivate accounts.
+    """
+    permission_classes = (permissions.IsAuthenticated, IsStaffUser)
+
+    def post(self, request, username):
+        """
+        POST /api/user/v1/accounts/{username}/deactivate/
+
+        Changes the 'is_active' attribute of a user.
+        """
+        user = User.objects.get(username=username)
+        user.is_active = False
```

Comment:
Unusable Password is the closest we get to "deleting" an account.  It will prevent the user from logging in to their account, which may or may not be the desired behavior. 

I agree that marking them inactive is likely not the correct approach.